Уф.

iPlayerOne opened the pull request.

Commit:
Initial

vadim-chistiakov commented on CryptoAssessment/ Networking/NetworkService.swift:
```diff
@@ -0,0 +1,95 @@
+
+import Foundation
+
+protocol NetworkService {
```
Сам NetworkService должен быть классом который умеет делать любые запросы. Потом создаешь допустим CoinService, в него инжектишь NetworkService

vadim-chistiakov commented on CryptoAssessment/Components/CoinList/СoinImageView.swift:
```diff
@@ -0,0 +1,52 @@
+    let imageUrl: String
+    
+    var body: some View {
+//        Image(systemName: "circle.fill")
```
Закомментированный код на ревью непринято отправлять

vadim-chistiakov commented:
Некоторые вьюхи получились жирными, можно часть бизнес логики уносить во ViewModels

vadim-chistiakov commented:
В целом норм, код читается понятно

Commit:
1. Изменил подход к DI контейнеру.
2. Попытался, где это было возможно создавать экщемпляры вью моделей во вью.
3. Разделил отвественность графика и остальных деталей.
4. Вместо ДТО теперь передаю в вью модели только айди объекта.

Commit:
Поменял концепцию  DI Контейнера
Где было возможно реализовал вью модели через создание экземпляра в вью.
Изменил способ передачи айди в связанные вью модели(больше это не весь ДТО а только стринг с айди.)
Разделил отвественность. Теперь у чарта есть вью модель которая работает с чартом, расчетами и анимациями а коиндитейл грузит только детали.
